### PR TITLE
Implementing linters

### PR DESCRIPTION
**Pull request by deniszh**
Fixing #365. Becoming quite tedious task, but it was fun.
Enabling `gofmt`, `goimports` and `gocritic`, together with others [golang-cilint defaults](https://golangci-lint.run/usage/linters/#enabled-by-default-linters) but disabling `errcheck` (it's too noisy IMO).
I avoided fixing unused struct and [S1012](https://staticcheck.io/docs/checks#S1012).

### CODE REVIEW

File: carbonserver/details.go [hunk 1]
```diff
@@ -7,19 +7,13 @@ import (
...
-	"unsafe"
 
 	"go.uber.org/zap"
 
 	"github.com/go-graphite/carbonzipper/zipper/httpHeaders"
-	protov2 "github.com/go-graphite/protocol/carbonapi_v2_pb"
 	protov3 "github.com/go-graphite/protocol/carbonapi_v3_pb"
 )
 
-func metricDetailsV2toV3(ptr *map[string]*protov2.MetricDetails) *map[string]*protov3.MetricDetails {
```
Comment (deniszh) [inline carbonserver/details.go:19]: /cc @Civil : could you please check - was `metricDetailsV2toV3` needed?

Comment (Civil) [reply carbonserver/details.go:19]: That's some leftovers from attempts to cheaply convert data from one format to another. So it should be safe to remove.

Comment (deniszh) [reply carbonserver/details.go:19]: Thanks for checking, @Civil !

**Review by avereha — APPROVED** [review]

**Pull request merged**